sugondat/chain/node/src/proposer.rs

```diff
@@ -89,8 +99,44 @@ impl<P: ProposerInterface<Block> + Send> ProposerInterface<Block> for BlockLimit
             // testing for detection of healthiness.
             parent_header.number == 0
         };
+        let exceeded_max_skipped_blocks = 'max_skipped: {
+            let maybe_last_relay_block_number = self.client.storage(
+                parent_header.parent_hash,
+                &sp_storage::StorageKey(sugondat_primitives::last_relay_block_number_key()),
+            );
 
-        if has_downward_message || has_go_ahead || has_transactions || first_block {
+            // If the state of the previous block or the last relay block number
+            // is not available, to be sure of not exceeding the max amount of
+            // skippable blocks, the block will be produced.
+            let last_relay_block_number = match maybe_last_relay_block_number {
+                Ok(Some(raw_data)) => match Decode::decode(&mut &raw_data.0[..]) {
+                    Ok(last_relay_block_number) => last_relay_block_number,
+                    Err(_) => break 'max_skipped true,
+                },
+                _ => break 'max_skipped true,
+            };
+            let relay_block_number = paras_inherent_data.validation_data.relay_parent_number;
+
+            // TODO: Change 3600 to 7200 (half a day)
+            // and `n_skipped_blocks = relay_parent_distance.saturating_sub(1)`
+            // when updating to asynchronous backing
+            // https://github.com/thrumdev/blobs/issues/166
+
+            // The accepted error is less than 10^(-2) for an expected
+            // maximum of 3600 skipped blocks (half a day)
+            let max_skipped_blocks = 3600;
```

Comment:
Should be `const`

Reply:
I have updated the code; now it is constant. Should it be directly dependent on some parameters specified in the pallet length fee adjustment? Or is it acceptable to keep those two values separated for now? Currently, the pallet relies on the assumption that the collator will produce at least one block for every 3600 skipped blocks. However, the pallet does not directly include information regarding this block count, as it was defined out of the code and the number of terms (that depends on the max number of skipped blocks and the max acceptable error) is defined in the runtime

Reply:
A constant is absolutely fine, as long as there is a comment justifying the decision